drivers/xen/dom0/domctl.c

```diff
@@ -272,15 +272,18 @@ int xen_domctl_max_vcpus(int domid, int max_vcpus)
 	return do_domctl(&domctl);
 }
 
-int xen_domctl_createdomain(int domid, struct xen_domctl_createdomain *config)
+int xen_domctl_createdomain(int *domid, struct xen_domctl_createdomain *config)
```

Comment:
It might be better to add 3d parameter like
`xen_domctl_createdomain(int domid, struct xen_domctl_createdomain *config, int *xen_domid)`
so it will be clearly optional.

Reply:
I don't think so, if Xen decides to replace the original domid value it means that the original one is not valid and does not point to any useful data. So keeping it would be detrimental as it leaves the possibility to accidentally use it for something.
Plus this signature is more similar to the libxl one.

Reply:
It would be responsibility of the caller. Proposed approach just will make impl. simpler.
use cases:
- create and forget with any id:  xen_domctl_createdomain(0, cfg, NULL);
- create and forget with specific  id:  xen_domctl_createdomain(id, cfg, NULL);
- create with any id and get xen_id:  xen_domctl_createdomain(0, cfg, &xen_id);
- create with specific if and check it:   xen_domctl_createdomain(id, cfg, &xen_id); if id != xen_id then panic.

I'll not insist. But note if you try to describe/document this API you might change your mind.